config/config-api/src/main/java/com/thoughtworks/go/config/Agent.java

```diff
@@ -0,0 +1,351 @@
+/*
+ * Copyright 2019 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.thoughtworks.go.config;
```

Comment:
> Since Agent is no longer a config object, it should not be in this package.

Yes! we will find the right home for `Agent` class.

Reply:
> Since Agent is no longer a config object, it should not be in this package.

Did you mean it can remain in the config module but under a different package or you meant moving completely away from the config module into some other module?